ipod-shuffle-4g.py

```diff
@@ -467,7 +520,7 @@ def set_master(self, tracks):
         if self.playlist_voiceover and (Text2Speech.valid_tts['pico2wave'] or Text2Speech.valid_tts['espeak']):
             self["dbid"] = hashlib.md5(b"masterlist").digest()[:8]
             self.text_to_speech("All songs", self["dbid"], True)
-        self["listtype"] = 1
+        self.listtype = PlaylistType.ALL_SONGS
```

Comment:
Why did you change from struct to class property?

Reply:
I created the class property `self.listtype` so I could set it to the enum value and then compare to the enum instead of converting to the `.value`. I'm not sure why I didn't do the same for the `PlaylistHeader`. I'd be inclined to make that one match with a new class property, rather than making this match by dropping it.